rfcs/datepicker-component.md

```diff
@@ -0,0 +1,161 @@
+
+* `DatePicker`
+* `StatefulDatePicker`
+* `StatefulDatePickerContainer`
+* `StyledRoot`
+* `StyledDateLabel`
+* `StyledDateInput`
+* `StyledMonthAndYearBar`
+* `StyledMonthSelect`
+* `StyledYearSelect`
+* `StyledMonthGrid`
+* `StyledMonthGridLabel`
+* `StyledMonthGridDay`
+* `StyledQuickSelect`
+
+## `DatePicker` API
+
+* `isRange: boolean` - Optional
+  If set, datepicker is to select 2 dates range, so there are two date input and `value` comes as Array of two dates, min and max. Otherwise only one date is selected. Default is `false`
+* `dateFormat: string` - Optional
+  Date format that is accepted by `Date.parse` function. RFC2822 tools.ietf.org/html/rfc2822#section-3.3 standard. Default is `DD/MM/YYYY`
+* `value: Date | Array<Date>` - Optional
+  Selected `Date`. If `isRange` is set, `value` is an array of 2  Date values".
+* `minDate: Date` - Optional
+  minimum date available for selection. Default is `Jan 01 1970`
+* `maxDate: Date` - Optional
+  maximum date available for selection. Default is current date + 3 years.
+* `disabled: boolean`:
+  Disable control from being changed
+* `monthRange: number`
+  The number of calendar months to show at once. Accepts number of visible months(mostly 1 or 2). Default is `1`
+* `overrides - Object:` - Optional
+
+```js
+             {  
+               Root: (props: {[string]: any}) => React$Node,
+               DateLabel: (props: {[string]: any}) => React$Node,
+               DateInput: (props: {[string]: any}) => React$Node,
+               MonthAndYearBar: (props: {[string]: any}) => React$Node,
+               MonthSelect: (props: {[string]: any}) => React$Node,
+               YearSelect: (props: {[string]: any}) => React$Nod,
+               MonthGrid: (props: {[string]: any}) => React$Node,
+               MonthGridLabel: (props: {[string]: any}) => React$Node,
+               MonthGridDay: (props: {[string]: any}) => React$Node,
+               QuickSelect: (props: {[string]: any}) => React$Node
+             }
+```
+
+* `overrides(continued):`
+  * `Root` - parent element of whole datepicker
+  * `DateLabel` - label above date input
+  * `DateInput` -  date input showing current selected date
+  * `MonthAndYearBar` - a bar\panel showing current selected month and year
+  * `MonthSelect` - select component to pick a month
+  * `YearSelect` - select component to pick a year
+  * `MonthGrid` - a grid of days in month
+  * `MonthGridLabel` - label for day of week in grid
+  * `MonthGridDay` - day number in a grid
+  * `QuickSelect` - set of buttons or any other components. Default contains quick selection of current week, month, year and other common ranges
+* `onChange: func(event, value)` - Optional
+  handler for events on trigger element, current selected date `value` (single or 2 dates)
+* `onMonthSelect: func(event, month)` - Optional
+  handler for events on trigger element, current month selected in Select dropdown. From 0 to 11.
+* `onYearSelect: func(event, year)` - Optional
```

Comment:
We'd also need an onChange handler when an actual date selected

Reply:
`onChange` is couple lines above